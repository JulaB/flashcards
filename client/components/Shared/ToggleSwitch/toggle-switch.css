.toggle-switch {
  width: 50px;
  height: 28px;
  cursor: pointer;
  position: relative;

  &__container::after,
  &__container::before {
    position: absolute;
    content: '';
    top: 50%;
    transform: translateY(-50%);
    cursor: pointer;
    transition: right 0.6s ease-in-out;
  }

  &__container::after {
    content: '';
    border-radius: 100%;
    width: 22px;
    height: 22px;
    line-height: 22px;
    text-align: center;
    font-size: var(--text-xl);
    margin-left: 3px;
    margin-right: 3px;
    cursor: pointer;
    background-color: var(--color-primary-dark);
  }

  &__container::before {
    width: 50px;
    height: 28px;
    border: 3px solid var(--color-primary);
    border-radius: 13px;
    background-color: var(--color-primary);
    cursor: pointer;
  }

  &__checkbox {
    position: absolute;
    left: 0;
    top: 0;
    outline: none;
    appearance: none;
    width: 100%;
    height: 100%;
    z-index: 1000;
  }

  &__checkbox:checked + .toggle-switch__container::before {
    background-color: var(--color-primary-dark);
    border-color: var(--color-primary-dark);
  }

  &__checkbox:checked + .toggle-switch__container::after {
    background-color: var(--color-primary);
    right: 0;
    content: '';
  }

  &__checkbox:focus + .toggle-switch__container::before {
    border-color: var(--color-border);
    outline-offset: 4px;
  }
}
