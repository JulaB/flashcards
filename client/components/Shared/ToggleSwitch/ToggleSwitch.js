import React from 'react';
import PropTypes from 'prop-types';
import './toggle-switch.css';

const ToggleSwitch = ({
  className,
  id,
  ...props
}) => (
  <label className={`toggle-switch ${className}`} htmlFor={id}>
    <input
      type="checkbox"
      className="toggle-switch__checkbox"
      id={id}
      role="switch"
      {...props}
    />
    <span area-hidden="true" className="toggle-switch__container"></span>
  </label>
);

ToggleSwitch.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
};

ToggleSwitch.defaultProps = {
  className: '',
  id: 'toggleSwitch',
};

export default ToggleSwitch;
