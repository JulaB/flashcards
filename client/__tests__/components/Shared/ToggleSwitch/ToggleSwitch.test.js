import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ToggleSwitch from 'components/Shared/ToggleSwitch/ToggleSwitch';

describe('<ToggleSwitch />', () => {
  it('renders the component', () => {
    const wrapper = shallow(<ToggleSwitch />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders component with custom props', () => {
    const wrapper = shallow(
      <ToggleSwitch
        id="customId"
        className="some-class"
        checked={true}
      />,
    );

    expect(wrapper.find('.toggle-switch')).toHaveLength(1);
    expect(wrapper.find('.toggle-switch').hasClass('some-class')).toBe(true);

    const checkbox = wrapper.find({ type: 'checkbox' });
    expect(checkbox.props().checked).toBe(true);
  });
});
