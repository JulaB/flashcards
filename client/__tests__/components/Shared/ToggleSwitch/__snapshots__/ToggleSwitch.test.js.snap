// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> renders the component 1`] = `
<label
  className="toggle-switch "
  htmlFor="toggleSwitch"
>
  <input
    className="toggle-switch__checkbox"
    id="toggleSwitch"
    role="switch"
    type="checkbox"
  />
  <span
    area-hidden="true"
    className="toggle-switch__container"
  />
</label>
`;
